### PR TITLE
fix bulk request (WP-912)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php [hunk 1]
```diff
@@ -97,16 +96,12 @@ public function bulkUpload(
                     $this->submissionManager->getSubmissionEntity($contentType, $currentBlogId, $id, $targetBlogId, $this->localizationPluginProxy);
                 $submission->setJobInfo($jobInfo);
                 $submission->setStatus(SubmissionEntity::SUBMISSION_STATUS_NEW);
-                $fileUri = $this->fileUriHelper->generateFileUri($submission);
-                $submission->setFileUri($fileUri);
                 $title = $this->getTitle($submission);
                 if ($title !== '') {
                     $submission->setSourceTitle($title);
                 }
+                $submission->setFileUri($this->fileUriHelper->generateFileUri($submission));
```
Comment (dimitrystd) [inline inc/Smartling/Services/ContentRelationsDiscoveryService.php:103]: I prefer avoiding this step for existing submissions. It should not harm, but it appears that you can rewrite the first lines.
```php
                $submission = $this->submissionManager->findTargetBlogSubmission($contentType, $currentBlogId, $id, $targetBlogId);
                if ($submission == null) {
                    $submission = $this->submissionManager->getSubmissionEntity($contentType, $currentBlogId, $id, $targetBlogId, $this->localizationPluginProxy);
                    $submission->setFileUri($this->fileUriHelper->generateFileUri($submission));
                }
```

**Review by dimitrystd — APPROVED** [review]

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php [hunk 2]
```diff
@@ -119,20 +114,27 @@ public function bulkUpload(
                 $carry[] = $item->ID;
                 return $carry;
             }, []);
-            $queueIds = $this->bulkUpload(
+            $queueIds[] = array_merge(...$this->bulkUpload(
```
Comment (PavelLoparev) [inline inc/Smartling/Services/ContentRelationsDiscoveryService.php:117]: `$queueIds` will look like `[0 => [...]]`?

Comment (PavelLoparev) [reply inc/Smartling/Services/ContentRelationsDiscoveryService.php:117]: Also I don't get why do you need array merge here?

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php [hunk 3]
```diff
@@ -119,20 +114,27 @@ public function bulkUpload(
...
                 $profile,
                 $targetBlogIds,
-                $batchUid,
                 false,
-            );
+            ));
         }
         if ($enqueue) {
-            $this->uploadQueueManager->enqueue(new IntegerIterator(array_merge(...$queueIds)), $batchUid);
+            $submissionIds = array_merge(...$queueIds);
```
Comment (PavelLoparev) [inline inc/Smartling/Services/ContentRelationsDiscoveryService.php:129]: Not sure why but `$queueIds` is already merged above. One more time?

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Services/ContentRelationsDiscoveryService.php:103]: Done

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php (hunk 2 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Services/ContentRelationsDiscoveryService.php:117]: Simplified code

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php (hunk 3 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Services/ContentRelationsDiscoveryService.php:129]: Yes, that was because queueIds was a flat array of arrays

**Review by PavelLoparev — APPROVED** [review]

**Review by dimitrystd — APPROVED** [review]

**Pull request merged**